src/engine/input/InputAction.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.Linq;
+using Godot;
+
+public class InputAction
+{
+    private ObservableCollection<InputEventWithModifiers> inputEvents;
+
+    private InputAction(string actionName, string displayName)
+    {
+        ActionName = actionName;
+        DisplayName = displayName;
+    }
+
+    public static ReadOnlyDictionary<string, IReadOnlyList<InputAction>> AllInputs { get; } = new ReadOnlyDictionary<string, IReadOnlyList<InputAction>>(new Dictionary<string, IReadOnlyList<InputAction>>
+    {
+        {
+            "Movement", ToReadonly(
+                new InputAction("g_move_forward",   "Move forward"),
```

Comment:
May I suggest that this information would be loaded from JSON (in simulation parameters)? That way you can edit this info without touching the code (which would make sense as you can configure new inputs in the godot project, which also doesn't need programming).

Reply:
Good idea